product_barcode_generator/models/product_template.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+import uuid
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.multi
+    def generate_barcode(self):
+        self.ensure_one()
+        if self.to_weight:
+            seq_internal_code = self.env.ref(
+                "product_barcode_generator.seq_ean_product_internal_ref"
+            )
+            bc = ""
+            if not self.default_code:
+                rule = self.env["barcode.rule"].search(
+                    [
+                        (
+                            "name",
+                            "=",
+                            "Price Barcodes (Computed Weight) 2 Decimals",
```

Comment:
I suspect that this was copied from upstream, but this seems terrible fragile. Does this barcode rule not have an xmlid?

No urgent need to change this if it was copied from upstream.

Reply:
Yes it was. I add a fixme in the code.